linux_stringio.cpp doesn't include <map>

## Description
```
[79/266] Building CXX object renderdoc/CMakeFiles/rdoc.dir/os/posix/linux/linux_stringio.cpp.o
FAILED: renderdoc/CMakeFiles/rdoc.dir/os/posix/linux/linux_stringio.cpp.o
/usr/bin/c++  -DRENDERDOC_EXPORTS -DRENDERDOC_PLATFORM_LINUX -DRENDERDOC_SUPPORT_EGL -DRENDERDOC_SUPPORT_VULKAN -DRENDERDOC_WINDOWING_WAYLAND -DRENDERDOC_WINDOWING_XCB -DRENDERDOC_WINDOWING_XLIB -I/opt/renderdoc/renderdoc -I/opt/renderdoc/renderdoc/3rdparty -std=c++11 -fstrict-aliasing -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Wno-unused-variable -Wno-unused-parameter -Wno-unused-result -Wno-type-limits -Wno-missing-field-initializers -Wno-unknown-pragmas -Wno-reorder -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-class-memaccess -Wimplicit-fallthrough=2 -Werror -fPIC -MD -MT renderdoc/CMakeFiles/rdoc.dir/os/posix/linux/linux_stringio.cpp.o -MF renderdoc/CMakeFiles/rdoc.dir/os/posix/linux/linux_stringio.cpp.o.d -o renderdoc/CMakeFiles/rdoc.dir/os/posix/linux/linux_stringio.cpp.o -c /opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:268:6: error: ‘map’ in namespace ‘std’ does not name a template type
  268 | std::map<rdcpair<wl_registry *, uint32_t>, wl_seat *> seatNames;
      |      ^~~
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:60:1: note: ‘std::map’ is defined in header ‘<map>’; did you forget to ‘#include <map>’?
   59 | #include <wayland-client.h>
  +++ |+#include <map>
   60 | #else
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:269:6: error: ‘map’ in namespace ‘std’ does not name a template type
  269 | std::map<wl_seat *, wl_keyboard *> seatKeyboard;
      |      ^~~
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:269:1: note: ‘std::map’ is defined in header ‘<map>’; did you forget to ‘#include <map>’?
  269 | std::map<wl_seat *, wl_keyboard *> seatKeyboard;
      | ^~~
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp: In function ‘void Keyboard::WaylandSeatCaps(void*, wl_seat*, uint32_t)’:
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:392:10: error: ‘seatKeyboard’ was not declared in this scope; did you mean ‘XGrabKeyboard’?
  392 |       if(seatKeyboard[seat])
      |          ^~~~~~~~~~~~
      |          XGrabKeyboard
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:405:7: error: ‘seatKeyboard’ was not declared in this scope; did you mean ‘XGrabKeyboard’?
  405 |       seatKeyboard[seat] = keyboard;
      |       ^~~~~~~~~~~~
      |       XGrabKeyboard
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:414:18: error: ‘seatKeyboard’ was not declared in this scope; did you mean ‘XGrabKeyboard’?
  414 |       keyboard = seatKeyboard[seat];
      |                  ^~~~~~~~~~~~
      |                  XGrabKeyboard
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp: In function ‘void Keyboard::WaylandRegistryAdd(void*, wl_registry*, uint32_t, const char*, uint32_t)’:
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:437:7: error: ‘seatNames’ was not declared in this scope
  437 |       seatNames[{reg, name}] = seat;
      |       ^~~~~~~~~
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp: In function ‘void Keyboard::WaylandRegistryRemove(void*, wl_registry*, uint32_t)’:
/opt/renderdoc/renderdoc/os/posix/linux/linux_stringio.cpp:445:13: error: ‘seatNames’ was not declared in this scope
  445 |   auto it = seatNames.find({reg, name});
      |             ^~~~~~~~~
```

## Steps to reproduce
1. `cmake /opt/renderdoc -DCMAKE_BUILD_TYPE=Relase -GNinja -DENABLE_GL=OFF -DENABLE_WAYLAND=ON -DENABLE_PYRENDERDOC=OFF -DENABLE_GLES=OFF -DENABLE_GL=OFF`
2. `ninja -j 16`

## Environment
* RenderDoc version: 8f909275a79118cc962d8910557569a2587dea7a
* Operating System: Debian Unstable
* Graphics API: N/A

GCC: 9.2.1 20200104 (Debian 9.2.1-22)
CMake: 3.16.20191223-git (self-compiled)

--- a/renderdoc/os/posix/linux/linux_stringio.cpp
+++ b/renderdoc/os/posix/linux/linux_stringio.cpp
@@ -57,6 +57,7 @@ struct xcb_connection_t;
 #if ENABLED(RDOC_WAYLAND)
 #include <linux/input.h>
 #include <wayland-client.h>
+#include <map>
 #else
 struct wl_display;
 struct wl_surface;

